lib/Jsrt/JsrtDebuggerObject.cpp

```diff
@@ -801,24 +801,30 @@ Js::JavascriptArray * JsrtDebugStackFrames::StackFrames(Js::ScriptContext * scri
     tempScriptContext != nullptr && tempScriptContext->IsScriptContextInDebugMode();
         tempScriptContext = tempScriptContext->next)
     {
-        Js::WeakDiagStack * framePointers = tempScriptContext->GetDebugContext()->GetProbeContainer()->GetFramePointers();
-        if (framePointers != nullptr)
+        Js::ProbeContainer* probeContainer = tempScriptContext->GetDebugContext()->GetProbeContainer();
+
+        // Only give frames which were preset when break was triggered as we might have other user (non-library) scripts executed after break
+        if (probeContainer->HaveFramePointers())
```

Comment:
I am not sure if I follow this. We create the stackframe for other scriptcontext only when they were asked for. So when do the other scriptcontext will have their frame pointers populated?